packages/block-editor/src/components/global-styles/hooks.js

```diff
@@ -75,22 +135,51 @@ export function useGlobalSetting( path, blockName, source = 'all' ) {
 	};
 
 	// Unlike styles settings get inherited from top level settings.
-	const resultWithFallback =
-		getSettingValueForContext( blockName ) ?? getSettingValueForContext();
+	const resultWithFallback = useMemo( () => {
+		if ( propertyPath ) {
+			return (
+				getSettingValueForContext( blockName, propertyPath ) ??
+				getSettingValueForContext( undefined, propertyPath )
+			);
+		}
+		const result = {};
+		VALID_SETTINGS.forEach( ( setting ) => {
```

Comment:
We can try, it will probably work for now but I believe that function is broken for properties that are supposed to be objects: for instance `layout`. If you merge two layouts, it's going to be merging the keys from both layouts but in reality it doesn't make sense, we should pick one layout or the other. In other words, I think we should do the opposite: Remove that merge function and use a static one based on "validKeys" or things like that.

Reply:
That's a great point. Does the PHP implementation already do that? Something to check.